staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -404,9 +407,9 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			// If this is "connection refused" error, it means that most likely apiserver is not responsive.
 			// It doesn't make sense to re-list all objects because most likely we will be able to restart
 			// watch where we ended.
-			// If that's the case wait and resend watch request.
+			// If that's the case begin exponentially backing off and resend watch request.
 			if utilnet.IsConnectionRefused(err) {
-				time.Sleep(time.Second)
+				<-r.initConnBackoffManager.Backoff().C()
```

Comment:
Do we need to reset this if it successfully connects?

I wouldn't want e.g. a second disconnection to wait 5 minutes before attempting to connect.

Reply:
I think that is handled by the backoff manager's `resetDuration` field. 

My reading of https://godoc.org/k8s.io/apimachinery/pkg/util/wait#NewExponentialBackoffManager thinks that after having not backed off for `resetDuration` it will start the backoff over again (i.e. only waiting `initBackoff` on the first iteration, increasing exponentially thereafter).

I could add a test case for this, but that'll 1) make the mocked out `WatchFunc` really messy and 2) just be testing that the backoff manager does what it says it does.

Reply:
Nope, if it's documented that way, that's fine, great, no need to double test here.